File: src/main/Java/boj_step3/boj25304.java
import java.io.*;
import java.util.StringTokenizer;

public class boj25304 {
    public static void main(String[] args) throws IOException {
        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter writer = new BufferedWriter(new OutputStreamWriter(System.out));
        int total = Integer.parseInt(reader.readLine());
        int n = Integer.parseInt(reader.readLine());
        int sumPrice = 0;
        while (n-- > 0) {
            StringTokenizer tokenizer = new StringTokenizer(reader.readLine());
            int price = Integer.parseInt(tokenizer.nextToken());
            int count = Integer.parseInt(tokenizer.nextToken());
            sumPrice += price * count;
        }
        writer.append((sumPrice == total) ? "Yes" : "No");
        writer.close();
    }
}

File: src/main/Cpp/boj_step3/boj25304.cpp
#include <iostream>
using namespace std;

int main() {
    int X, N, a, b, total = 0;
    cin >> X >> N;
    for (int i = 0; i < N; i++) {
        cin >> a >> b;
        total += (a * b);
    }
    cout << ((X == total) ? "Yes" : "No");
}